src/hsmm_mvpy/models.py

```diff
@@ -1090,26 +1090,24 @@ def distribution_pmf(self, shape, scale, location):
         shape : float
             shape parameter
         scale : float
-            scale parameter
-        location: float
-            location parameter      
+            scale parameter     
         Returns
         -------
         p : ndarray
             probabilty mass function for the distribution with given scale
         '''
         if scale == 0:
             warn('Convergence failed: one stage has been found to be null')
-        p = self.cdf(np.arange(1, self.max_d+1), shape, scale=scale, loc=location)
-        #Location is at least = to 1 given that range starts at 0, desirable as these are durations
-        p = np.diff(p, prepend=0)#going to pmf
+        p = self.pdf(np.arange(self.max_d)+1, shape, scale=scale)
+        p[:location] = 0
+        p[np.isnan(p)] = 0 #remove potential nans
```

Comment:
I never checked, does this happen regularly?

Reply:
No, again only in an edge case where it tries to define super narrow pdfs. This fixes it, but then makes the likelihood go to -inf (otherwise you get errors)

Reply:
We can leave it to avoid the errors but then make a more clear warning, but not necessary right now